SoftLayer/managers/ordering.py

```diff
@@ -471,3 +473,22 @@ def package_locations(self, package_keyname):
 
         regions = self.package_svc.getRegions(id=package['id'], mask=mask)
         return regions
+
+    def get_location_id(self, location):
+        """Finds the location ID of a given datacenter
+
+        This is mostly used so either a dc name, or regions keyname can be used when ordering
+        :param str location: Region Keyname (DALLAS13) or datacenter name (dal13)
+        :returns: integer id of the datacenter
+        """
+
+        mask = "mask[id,name,regions[keyname]]"
+        if match(r'[a-zA-Z]{3}[0-9]{2}', location) is not None:
+            search = {'name': {'operation': location}}
+        else:
+            search = {'regions': {'keyname': {'operation': location}}}
+        datacenter = self.client.call('SoftLayer_Location', 'getDatacenters', mask=mask, filter=search)
+        # [{'id': 1854895, 'name': 'dal13', 'regions': [{'keyname': 'DALLAS13'}]}]
```

Comment:
I'm not a fan of putting example results like this because they can get out of date quickly. The corresponding fixture is a good way to discover the result structure that is actually used instead IMHO.